### DISCUSSION TITLE
Exporting Getters and Actions type to allow external definitions

### DISCUSSION THREAD

**Discussion by jfrs** [vuejs/pinia#674] [Ideas]
When using Vuex, if my store gets too big I can split everything into different objects/files and keep the types:

```
  getters?: GetterTree<MyState, MyRootState>;
  actions?: ActionTree<MyState, MyRootState>;
```

Pinia has complex types for the getters and actions and it would be nice to be able to import a single type to do the same if possible.

I have tried to do something like this but couldn't get it to work:

```
export type PiniaGetters<S> = {} & ThisType<UnwrapRef<S> & StoreWithGetters<{}> & PiniaCustomProperties> & GettersTree<S>;
export type PiniaActions<Id extends string, S, G> = {} & ThisType<{} & UnwrapRef<S> & StoreWithState<Id, S, G, {}> & StoreWithGetters<G> & PiniaCustomProperties>;
```

I'm loving everything else about this project, it's made jump to Vue 3, so thank you!

**Comment by posva** [comment]
Sorry for the late reply. Types are quite complex because the goal is to not write manual declarations, intead let pinia infer the types for you, staying closer to just js, even in ts.
